File: public/actions/python/encryptDataInKms.py
async def encryptDataInKms(kmsServerUrl: string, apiKey: string, public_key_uid: string, plaintext: string, policy: string):
  """Encrypt data in KMS"""
  kms_client = KmsClient(server_url=kmsServerUrl, api_key=apiKey)
  ciphertext = await kms_client.cover_crypt_encryption(
    policy,
    plaintext,
    public_key_uid,
  )

File: public/actions/javascript/encryptDataInKms.ts
import { KmsClient } from "cloudproof_js";
import { BACKEND_URL } from "./backendConfig";

//
// Encrypt data in KMS
//
export const encryptDataInKms = async (
  clearData: string,
  kmsToken: string,
  accessPolicy: string,
  masterPublicKeyUid: string
): Promise<Uint8Array> => {
  const client = new KmsClient(BACKEND_URL, kmsToken);
  const data = new TextEncoder().encode(clearData);
  const encryptedData = await client.coverCryptEncrypt(masterPublicKeyUid, accessPolicy, data);
  return encryptedData;
};

export const bulkEncryptInKms = async (
  clearData: string[],
  kmsToken: string,
  accessPolicy: string,
  mpkID: string
): Promise<Uint8Array[]> => {
  const client = new KmsClient(BACKEND_URL, kmsToken);
  const data = clearData.map((clear) => new TextEncoder().encode(clear));
  const response = await client.coverCryptBulkEncrypt(mpkID, accessPolicy, data);
  return response;
};
